[zhanyang01] iP

zhanyang01 opened the pull request:
# DukeKing

> _"It's not always that we need to do more but rather we need to focus on less_" -Nathan W.Morris, author and finance expert 
[source](https://www.tameday.com/motivational-quotes-for-work-productivity/)

Having too much work to do as a student 🤢  and worry about forgetting deadlines for submissions 💀 , fret not, as DukeKing is here to save the day and allow us to reduce our stress levels. 😀 

With DukeKing, you can focus on the things that truly matter!!!! 

### Important things to do ❗ ❗ ❗ 
1. ~~Remembering task by memory~~ 
2. Doing actual work

In addition, it is, 
* text-based
* easy to learn
* ~~FAST~~ SUPER FAST to use!!

All you need to do to use this amazing service is to,
1. download it from [here](https://github.com/zhanyang01/ip)
2. double-click on the .jar file
3. add tasks
4. let it manage tasks

and it is **FREE**!!! 👍 💯 

Currently it has, 
- [x] Managing tasks
- [ ] Managing datelines(coming soon)
- [ ] Reminders(coming soon) 

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

zhanyang01 committed:
Add level-0

zhanyang01 committed:
add Level-1

zhanyang01 committed:
add Level-2, fix lines in level 1

zhanyang01 committed:
add Level-3

zhanyang01 committed:
add Level-4

zhanyang01 committed:
add A-TextUiTesting

zhanyang01 committed:
add Level-5

zhanyang01 committed:
add Level-6

zhanyang01 committed:
add A-Enums, fix input and expected files

zhanyang01 committed:
Fix Level-5

zhanyang01 committed:
Rearrange Files

zhanyang01 committed:
Add Level-7

zhanyang01 committed:
Merge branch 'branch-level-7'

zhanyang01 committed:
Add Level-8

zhanyang01 committed:
Merge branch 'branch-Level-8'

zhanyang01 committed:
Add A-MoreOOP

zhanyang01 committed:
Add File

zhanyang01 committed:
Merge branch 'branch-A-MoreOOP'

zhanyang01 committed:
Add A-Packages

zhanyang01 committed:
Merge branch 'branch-A-Packages'

zhanyang01 committed:
Add more packages

zhanyang01 committed:
Merge branch 'branch-A-Packages'

zhanyang01 committed:
Merge branch 'add-gradle-support' of https://github.com/zhanyang01/ip into branch-A-Gradle

zhanyang01 committed:
Add A-Gradle

zhanyang01 committed:
Merge branch 'branch-A-Gradle'

zhanyang01 committed:
Add A-JUnit

zhanyang01 committed:
Add A-Jar

zhanyang01 committed:
Edit A-Jar

zhanyang01 committed:
Merge branch 'branch-A-Jar'

zhanyang01 committed:
Code reuse for learning purposes

zhanyang01 committed:
Add A-JavaDoc

zhanyang01 committed:
Add A-CodingStandard

zhanyang01 committed:
Add Level-9

zhanyang01 committed:
Merge branch 'branch-A-JavaDoc'

zhanyang01 committed:
Add dataTask.txt

zhanyang01 committed:
Add files

zhanyang01 committed:
Fix git merge conflict

zhanyang01 committed:
Merge branch 'branch-A-CodingStandard'

zhanyang01 committed:
Merge branch-Level-9

Fix git merge conflict

laurenlim2112 commented on src/main/java/duke/Tasks/Task.java:
```diff
@@ -0,0 +1,177 @@
+     * @param sDate is the string of the date
+     * @return a localDate object
+     */
+    public static LocalDate convertDatePlease(String sDate) {
```
Perhaps this method name should be phrased more as a verb?

GohTengFong replied:
I agree. I suggest removing the "Please" and phrasing it such as "convertDateFromStringToObj"?

laurenlim2112 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,121 @@
+     * @throws UnknownCommandException if asked commands that the bot do not
+     *                                 understand
+     */
+    public static boolean isExitOrContinue(String command,
```
This method might be a bit long; perhaps you could consider splitting it into 2 functions, with 1 function responsible for carrying out the command and another responsible for checking if the command is the "bye" command?

laurenlim2112 commented on src/main/java/duke/UI.java:
```diff
@@ -0,0 +1,110 @@
+     * @param task Take in the task to be marked
+     */
+    public void markTask(Task task) {
+        String markingTask = "Nice! I've marked this task as done:";
```
Should this variable be named with a verb or a noun?

laurenlim2112 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,121 @@
+
+import java.io.IOException;
+
+import duke.exceptions.EmptyDetailsOfTaskError;
```
I like how you consistently list your imported classes explicitly. Good job!

laurenlim2112 reviewed (commented):
Overall, your code is quite clean, except for some areas with room for improvement in terms of readability. Well done for the iP so far!

zhanyang01 committed:
Add A-CheckStyle

zhanyang01 committed:
Merge branch 'branch-A-CheckStyle'

GohTengFong commented on src/main/java/Duke.java:
```diff
@@ -2,6 +2,10 @@
 import java.util.ArrayList;
 public class Duke {
+
+    public enum Activity {
```
Should enum be in all caps?

GohTengFong commented on src/main/java/duke/Commands.java:
```diff
@@ -0,0 +1,16 @@
+ * List of all possible commands, used the idea of enums to list
+ * all the possible commands
+ */
+public enum Commands {
```
Should enum be in all caps?

GohTengFong commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,83 @@
+ * The main file that the application runs
+ */
+public class Duke {
+    protected UI helper;
```
Perhaps a more intuitive attribute name here?

GohTengFong commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,121 @@
+        String[] splittedCommand = command.split(" ");
+        String commandType = splittedCommand[0];
+        switch (commandType) {
+            case "bye":
```
Should each "case" be at the same indentation level as "switch" based on coding standards?

GohTengFong commented on src/main/java/duke/Exceptions/DukeException.java:
```diff
@@ -0,0 +1,27 @@
+ * Exceptions of Duke object, errors which duke is likely to encounter and what is done
+ * to catch and deal with the errors
+ */
+public class DukeException extends Throwable {
```
I love how you have many different specific Duke execeptions that extend "DukeExeception".

GohTengFong reviewed (commented):
Similar to Lauren, I think your code is very clean! I love how all your files are split according to specific packages! :D

zhanyang01 committed:
Add GUI, Level 10 not done yet

zhanyang01 committed:
Add Level-10

zhanyang01 committed:
Merge branch 'branch-Level-10'

zhanyang01 committed:
Fix Level-10

GUI now closes on bye command

GUI was not closing on bye command.

GUI that closes on bye command is more user-friendly as bye command
is supposed to exit the program and terminate the problem, closing
the GUI does that job.

Add Program.exit() in MainWindow solves it

zhanyang01 committed:
Add A-Assertions

zhanyang01 committed:
Refactor Command classes

Parser now is shorter and it fulfills SLAP as it
only has the functionality to return a command, the
functionality to return a string with the command is now
at the command classes.

Parser was very long and does not fulfill SLAP.

This will make the code more readable and improve the code quality.
This is part of A-CodeQuality.

zhanyang01 committed:
Add A-CodeQuality

zhanyang01 committed:
Merge pull request #1 from zhanyang01/branch-A-Assertions

Add A-Assertions

zhanyang01 committed:
Merge branch 'master' of https://github.com/zhanyang01/ip

* 'master' of https://github.com/zhanyang01/ip:
  Add A-Assertions

zhanyang01 committed:
Merge branch 'master' into branch-A-CodeQuality

zhanyang01 committed:
Merge pull request #2 from zhanyang01/branch-A-CodeQuality

Add A-CodeQuality

zhanyang01 committed:
Merge branch 'master' of https://github.com/zhanyang01/ip

* 'master' of https://github.com/zhanyang01/ip:
  Add A-CodeQuality
  Refactor Command classes

zhanyang01 committed:
Fix A-CodeQuality

zhanyang01 committed:
Merge branch 'branch-A-CodeQuality' of https://github.com/zhanyang01/ip into branch-A-CodeQuality

* 'branch-A-CodeQuality' of https://github.com/zhanyang01/ip:
  Add A-Assertions

zhanyang01 committed:
Merge branch 'branch-A-CodeQuality'

zhanyang01 committed:
Fix A-CodeQuality

zhanyang01 committed:
Add BCD-Extension, Fix GUI Welcome and Bye

zhanyang01 committed:
Merge branch 'branch-BCD-Extension'

zhanyang01 committed:
Add more details to C-Help

zhanyang01 committed:
Push case-insensative changes

zhanyang01 committed:
Fix case sensativity

zhanyang01 committed:
Fix bye command bug

Originally in text-based-ui, need to say bye 2 times to end

Now it works as normal that only saying bye once will end the
program

zhanyang01 committed:
Fix createTask method in duke.Task

Bug: will not accept tasks with more than 1 word
event and deadline will not return the expected deadlines or
expected events

Now it is able to accept tasks with multi-word descriptions

zhanyang01 committed:
Add A-Personality

zhanyang01 committed:
Edit Add A-Personality

zhanyang01 committed:
Edit A-UserGuide

zhanyang01 committed:
Edit A-UserGuide

zhanyang01 committed:
Edit A-UserGuide

zhanyang01 committed:
Add A-UserGuide

zhanyang01 committed:
Edit A-UserGuide

zhanyang01 committed:
Edit A-UserGuide

zhanyang01 committed:
Edit A-UserGuide

zhanyang01 committed:
Edit A-UserGuide

zhanyang01 committed:
Edit UI to fulfill SLAP

zhanyang01 committed:
Add A-Release

zhanyang01 committed:
Fix iP code quality

zhanyang01 committed:
Fix iP code quality

zhanyang01 committed:
Fix iP code quality

zhanyang01 committed:
Update A-Release

zhanyang01 committed:
Update A-Release

zhanyang01 committed:
Fix bugs in commands

zhanyang01 committed:
Fix code quality
